lib/spack/spack/config.py

```diff
@@ -306,6 +307,25 @@ def write_section(self, section):
     def __repr__(self):
         return '<InternalConfigScope: %s>' % self.name
 
+    @staticmethod
+    def _process_dict_keyname_overrides(data):
+        """Turn a trailing `:' in a key name into an override attribute."""
+        result = {}
+        for sk, sv in iteritems(data):
+            if sk.endswith(':'):
+                key = syaml.syaml_str(sk[:-1])
+                key.override = True
+            else:
+                key = sk
+
+            if isinstance(sv, dict):
+                result[key]\
+                    = InternalConfigScope._process_dict_keyname_overrides(sv)
+            else:
```

Comment:
Lists may contain dictionaries. This may require all the nuances from the `_mark_overrides` function that was removed in https://github.com/spack/spack/pull/11602.

Reply:
@scheibelp I don't think it does, because of the rules for lists: either we merge lists, in which case list elements can never override each other and the point is moot; or one list overrides the other _in toto_ and lower-level override flags are meaningless. Am I missing something?